rust/blockstore/src/arrow/provider.rs

```diff
@@ -34,8 +34,8 @@ impl ArrowBlockfileProvider {
     pub fn new(
         storage: Storage,
         max_block_size_bytes: usize,
-        block_cache: Cache<Uuid, Block>,
-        sparse_index_cache: Cache<Uuid, SparseIndex>,
+        block_cache: Box<dyn Cache<Uuid, CachedBlock>>,
```

Comment:
Does this mean that all cache accesses will use to_bytes/from_bytes? That seems not super great since its in the hot path?